Source/missiles.cpp

```diff
@@ -1375,12 +1375,18 @@ void AddLightningWall(Missile &missile, const AddMissileParameter &parameter)
 	UpdateMissileVelocity(missile, parameter.dst, 16);
 	missile._miAnimFrame = GenerateRnd(8) + 1;
 	missile._mirange = 255 * (missile._mispllvl + 1);
-	if (missile._misource < 0) {
+	switch (missile.sourceType()) {
+	case MissileSource::Trap:
 		missile.var1 = missile.position.start.x;
 		missile.var2 = missile.position.start.y;
-	} else {
-		missile.var1 = Players[missile._misource].position.tile.x;
-		missile.var2 = Players[missile._misource].position.tile.y;
+		break;
+	case MissileSource::Player: {
+		Player &player = *missile.sourcePlayer();
+		missile.var1 = player.position.tile.x;
+		missile.var2 = player.position.tile.y;
+	} break;
+	case MissileSource::Monster:
+		break;

```

Comment:
Perhaps it should raise an error instead?

Reply:
exceptions are discouraged, but you could add a debug assert to document the expectation that this is never hit.

Reply:
Something like this?

```c++
	case MissileSource::Monster:
		assert(false); // Monster source not supported
		break;
```

Reply:
yeah that'd work, maybe make the comment "Monsters aren't expected to cast lightning wall"

Reply:
Done. Thanks.